quickjspp.hpp

```diff
@@ -1626,6 +1632,25 @@ class Context
     /** Callback triggered when a Promise rejection won't ever be handled */
     std::function<void(Value)> onUnhandledPromiseRejection;
 
+    /** Data type returned by the moduleLoader function */
+    struct ModuleData {
+        std::optional<std::string> source, url;
+        ModuleData()
+          : source(std::nullopt), url(std::nullopt)
+        {}
+        ModuleData(std::string source)
+          : source(std::move(source)), url(std::nullopt)
+        {}
+        ModuleData(std::string url, std::string source)
+          : source(std::move(source)), url(std::move(url))
+        {}
```

Comment:
Not quite. The default constructors would swap the order of `url` and `source` in the 2 arguments case.
Also the default constructor would expect `std::optional` instead of `std::string`. I think that would limit implicit conversions (e.g., from `const char *` to `std::optional<std::string>`)